### DISCUSSION TITLE
Calculate cumulative sum of the column

### DISCUSSION THREAD

**Discussion by ShivaniSK** [finos/TimeBase-CE#232] [Q&A] [answered]
Hi @alex-karpovich ,
We have data stored in timebase as below 

```
**Column 1**
90
2
4
1
```

Is there a way to query timebase in a way to get cumulative sum of Column 1 in a new column ? 
We want a new column2 which give the cumulative sum of the values in Column 1 like below

```
Column1   Column2
90          90
2           92
4           96
1           97
```

Regards,
Shivani

**Comment by alex-karpovich** [comment, marked as answer]
@ShivaniSK sorry for long delay, it's possible for 6.1.14

```
select running sum(column1), column1 as column2 from stream
```

**Reply by ShivaniSK** [reply]
Hi @alex-karpovich ,
We need cumulation of data in below output format. Can you please suggest the QQL for the same.

**input data**

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 15">
<meta name=Originator content="Microsoft Word 15">
<link rel=File-List
href="file:///C:/Users/kanashia/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

<link rel=themeData
href="file:///C:/Users/kanashia/AppData/Local/Temp/msohtmlclip1/01/clip_themedata.thmx">
<link rel=colorSchemeMapping
href="file:///C:/Users/kanashia/AppData/Local/Temp/msohtmlclip1/01/clip_colorschememapping.xml">

<style>

</style>

</head>

<body lang=EN-US style='tab-interval:.5in;word-wrap:break-word'>




Epochtime | Symbol | Level | Column1
-- | -- | -- | --
12:01 | ABC | 1 | 90
12:01 | ABC | 2 | 2
12:01 | ABC | 3 | 2
12:02 | ABC | 1 | 4
12:02 | ABC | 2 | 1
12:02 | ABC | 3 | 2




</body>

</html>


**output data**

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 15">
<meta name=Originator content="Microsoft Word 15">
<link rel=File-List
href="file:///C:/Users/kanashia/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

<link rel=themeData
href="file:///C:/Users/kanashia/AppData/Local/Temp/msohtmlclip1/01/clip_themedata.thmx">
<link rel=colorSchemeMapping
href="file:///C:/Users/kanashia/AppData/Local/Temp/msohtmlclip1/01/clip_colorschememapping.xml">

<style>

</style>

</head>

<body lang=EN-US style='tab-interval:.5in;word-wrap:break-word'>




Epochtime | Symbol | Level | Column1 | Cumulative   Sum
-- | -- | -- | -- | --
12:01 | ABC | 1 | 90 | 90
12:01 | ABC | 2 | 2 | 92
12:01 | ABC | 3 | 2 | 94
12:02 | ABC | 1 | 4 | 4
12:02 | ABC | 2 | 1 | 5
12:02 | ABC | 3 | 2 | 7




</body>

</html>

**Comment by alex-karpovich** [comment]
@ShivaniSK 
```
select running sum{} (column1) as 'Cumulative Sum', column1 as column2 from stream RESET OVER TIME(1m)
```

link to the description of these functions

https://kb.timebase.info/docs/development/qql/QQL%205.5/qql-tut-functions#stateful-functions

**Reply by ShivaniSK** [reply]
Thanks @alex-karpovich
